lib/purge-users-using-local-file.rb

```diff
@@ -0,0 +1,79 @@
+require 'zendesk_api'
+require 'rest-client'
+require 'json'
+require 'date'
+
+@client = ZendeskAPI::Client.new do |config|
+  config.url = ENV['ZENDESK_URL']
+  config.username = ENV['ZENDESK_USER_EMAIL']
+  config.password = ENV['ZENDESK_USER_PASSWORD']
+  config.retry = true
+end
+
+# Setup vars
+
+diag = "true"
+url = "#{ENV['ZENDESK_URL']}/deleted_users/"
+day = Date.today.next_day
+lastyear = day - 365
+
+input_file = File.open "data/soft_deleted_users.json"
```

Comment:
Opened file without corresponding close. Consider, more idiomatically:

```ruby
File.open "data/soft_deleted_users.json" do |input_file|
  file_data = JSON.load input_file
end
```

Reply:
@blairboy362  Hey Daniel, I've substituted your code blob in place of mine and tested. The script fails with: filename.rb:30:in `<main>': undefined local variable or method `file_data' for main:Object (NameError)

<<< snip >>>
File.open "data/soft_deleted_users_test.json" do |input_file|
  file_data = JSON.load input_file
end

file_data.each do |user|

  # Extract some fields
  updated_at = user["updated_at"]
  last_login_at = user["last_login_at"]
blah......

Reply:
I did have a lot of trouble to get this section to function correctly, even though there is no 'close' statement.

Reply:
Let's spend some time tomorrow morning pairing on this so we can sort it. Shouldn't take long.